### PR TITLE
build(tsconfig): emit esm output

### LINKED ISSUE #61: Feature Request: [Feature] tree-shakable dist output support
**Description / Use Case for a Feature**

This is feature request to support `module` field in package.json allows bundler can pick up tree-shakable code. Whole size of tslog itself is relatively small enough, but I hope tslog could expose import paths to skip few deps like `source-map` as well to squeeze size: it may possible by having export site in top level export (index.ts), but should be esm syntax to allow tree shaking.

Below's size composition for tslog which source-map takes more than half.
image

I do not have a strong opinion if it should be true esm or cjs output with esm syntax only - but given native esm support is not wide enough yet, native esm could wait longer. 

Happy to try PR if you don't mind.

### PR DESCRIPTION
**Pull request by kwonoj**
closes #61

This PR attempts to address #61 with few changes

- emit `dist/esm` with `module:esnext`, mark `module` property in package.json: This'll allow bundler will look for esm module when it tries to bundle
- cjs runtime now placed under `dist/cjs`: since package.json points to moved location, using plain cjs should work as is.
- mark `sideEffect` in package.json: dependency `source-map-support` and its inner deps does not export esm, bundler cannot detect if it's side effect free or not. Marking it explicitly since source-map-support's interface requires to call `install()` to trigger side effect, and tslog doesn't use it regardless.
- exposing `LoggerWithoutCallSite` interface: This is exactly same as current `Logger`, only difference is it doesn't call `wrapCallSite`. This is to support cases where prod bundle ships without source map and callsite wrapping isn't required. The reason it doesn't take runtime flag (`ISettings`) is bundler cannot analyze tree shaking if it becomes runtime flag - it'll consider all import will be used and will bundle all dependencies.

Overall, this PR shouldn't cause breaking changes: but also it introduces few notable changes ( and changes code organization).

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -80,8 +81,10 @@
     "source-map-support": "^0.5.19"
   },
   "files": [
-    "dist"
+    "dist",
+    "src"
```
Comment (kwonoj) [inline package.json:85]: declarationmap requires src to locate source from type definitions.

**Comment by kwonoj** [comment]
With this change, if use LoggerWithoutCallsite bundler will picks up these amount of bytesize only:

image

**Comment by terehov** [comment]
Looks really good, thanks. Since this is quite a big change (as well as an advantage) I guess it would be good to mention it in README. It would be amazing if you could add some lines mentioning tree shaking as well as this new version without callsites. Thank you!

**Comment by kwonoj** [comment]
tried update documentation, not sure if this is ok to go.

**Comment by terehov** [comment]
@kwonoj It's quite a big change, yet doesn't break anything. What do you think, should that be a new major release or just a minor?

**Comment by kwonoj** [comment]
As long as I didn't make any mistakes, this shouldn't cause any breaking so I'd consider semver minor. But given change's large, if it's my lib I'd publish beta tagged version first and leave it for a while.

**Comment by terehov** [comment]
Thant's a really good idea. It's a bit late here already, so I didn't think of that 👍

**Comment by terehov** [comment]
https://github.com/fullstack-build/tslog/releases/tag/v2.12.0-0

**Pull request merged**